### ISSUE TITLE
transmission broken in oe7

### ISSUE THREAD

**Issue by ZeeD** [OpenELEC/unofficial-addons#162]
the transmission-daemon is not working anymore after the update OE.

# ldd /storage/.kodi/addons/service.downloadmanager.transmission/bin/*   | grep not
        libcrypto.so.32 => not found
        libcrypto.so.32 => not found
$       not a dynamic executable

**Comment by odony** [comment]
The cleanest solution is to rebuild the *transmission* addon for OpenELEC  7, in order to link it to libcrypto **35** as included with OpenELEC 7.

These are the steps I followed to cross-compile it for RPi2/3 on my **Ubuntu** system:

0. Install some dependencies as [recommended by the OE doc](http://wiki.openelec.tv/index.php/Compile_from_source):
`sudo apt-get install git make libxml-parser-perl cmake`
1. Checkout OE, and switch to version 7:
```
git clone https://github.com/OpenELEC/OpenELEC.tv.git
cd OpenELEC.tv
git checkout -b 7.0.0
```
2. Get the unofficial addon repo (this repo)
`git clone https://github.com/OpenELEC/unofficial-addons.git packages/unofficial`

3. (until fixed) Patch the download URL for transmission or apply [the patch](https://patch-diff.githubusercontent.com/raw/OpenELEC/unofficial-addons/pull/160.patch) from #160:
```
cd packages/unofficial
curl https://patch-diff.githubusercontent.com/raw/OpenELEC/unofficial-addons/pull/160.diff | patch -p1
cd ../..
```
4. Cross-compile transmission for the right arch (PROJECT=RPi2):
`PROJECT=RPi2 ARCH=arm ./scripts/create_addon transmission`
5. Wait a long time (the script will install missing deps for you) and perhaps answer yes to a few questions...
6. Find the zipfile for the transmission addon in `target/addons/7.0/RPi2/arm/service.downloadmanager.transmission`

I suppose someone ought to do this for all _unofficial_ addons (and all supported architectures) and push the builds in the public repository of unofficial addons. Not sure who to ask, though ;-)

**Comment by maxonly1** [comment]
@odony odony, will these steps also work for making rTorrent work again on OE8?

thnx

**Comment by odony** [comment]
@maxonly1 probably not unless they are adapted. I couldn't manage to build rTorrent for OE8 after a quick test. Could be related to the other issues with broken dependencies in the latest rTorrent builds.

**Comment by maxonly1** [comment]
@odony  , yeah i tried to adapt the lines. I starts compiling but when it's finished there's no Openelec.tv/target folder. maybe it wasnt completly done. What errors did you get?
